badge_server/main.py

```diff
@@ -137,24 +138,46 @@ def get_highest_status(cls, statuses: Iterable[enum.Enum]):
 }
 
 
-def _get_supported_versions(package_name: str) -> List[int]:
-    """Gets the given package's supported python versions
+def _get_missing_details(package_names: List[str],
+                         results: Iterable[CompatibilityResult]
+                         ) -> Optional[str]:
+    """Gets the details for any missing data (if there is any)
 
     Args:
-        package_name: The name of the package to look up, e.g. "tensorflow".
+        package_names: A list of length 1 or 2 of the package name(s) to look
+            up, e.g. ["tensorflow"], ["tensorflow", "opencensus"].
+        results: A list of length 1 or 2 of the `CompatibilityResults` for the
+            given package(s). Results should come from datastore.
 
     Returns:
-        A list of supported python versions.
+        None if there is no missing data; a description of the version(s)
+        missing otherwise, e.g. "Missing data for python version(s): 2".
     """
-    supported = []
-    if not compat_utils._is_package_in_whitelist([package_name]):
-        return supported
-    unsupported_package_mapping = configs.PKG_PY_VERSION_NOT_SUPPORTED
-    for version in (2, 3):
-        unsupported_packages = unsupported_package_mapping[version]
-        if package_name not in unsupported_packages:
-            supported.append(version)
-    return supported
+    bad_list_length = 'package_names: Expected length of 1 or 2, got {}'
```

Comment:
I still don't like these function-level constants ;-)

Reply:
changed it